searchcore/src/tests/proton/attribute/attribute_test.cpp

```diff
@@ -917,38 +905,35 @@ struct StructMapFixture : public StructFixtureBase
         doc->setValue(_structMapField, s);
         return doc;
     }
+
     void checkAttrs(uint32_t lid, int32_t expValue, const std::map<int32_t, int32_t> &expMap) {
         auto valueAttr = _m->getAttribute("value")->getSP();
         auto mapKeyAttr = _m->getAttribute("map.key")->getSP();
         auto mapValueAttr = _m->getAttribute("map.value.value")->getSP();
-        EXPECT_EQUAL(expValue, valueAttr->getInt(lid));
+        EXPECT_EQ(expValue, valueAttr->getInt(lid));
         attribute::IntegerContent mapKeys;
         mapKeys.fill(*mapKeyAttr, lid);
         attribute::IntegerContent mapValues;
         mapValues.fill(*mapValueAttr, lid);
-        EXPECT_EQUAL(expMap.size(), mapValues.size());
-        EXPECT_EQUAL(expMap.size(), mapKeys.size());
+        EXPECT_EQ(expMap.size(), mapValues.size());
+        EXPECT_EQ(expMap.size(), mapKeys.size());
         size_t i = 0;
         for (const auto &expMapElem : expMap) {
-            EXPECT_EQUAL(expMapElem.first, mapKeys[i]);
-            EXPECT_EQUAL(expMapElem.second, mapValues[i]);
+            EXPECT_EQ(expMapElem.first, mapKeys[i]);
+            EXPECT_EQ(expMapElem.second, mapValues[i]);
             ++i;
         }
     }
 };
 
-TEST_F("require that update with doc argument updates struct field attributes (map)", StructMapFixture)
+TEST_F(StructMapWriterTest, update_with_doc_argument_updates_struct_field_attributes)
 {
-    auto doc = f.makeDoc(10,  {{1, 11}, {2, 12}});
-    f.put(10, *doc, 1);
-    TEST_DO(f.checkAttrs(1, 10, {{1, 11}, {2, 12}}));
-    doc = f.makeDoc(20, {{42, 21}});
-    f.update(11, *doc, 1, true);
-    TEST_DO(f.checkAttrs(1, 10, {{42, 21}}));
+    auto doc = makeDoc(10,  {{1, 11}, {2, 12}});
+    put(10, *doc, 1);
+    checkAttrs(1, 10, {{1, 11}, {2, 12}});
+    doc = makeDoc(20, {{42, 21}});
+    update(11, *doc, 1, true);
+    checkAttrs(1, 10, {{42, 21}});
 }
 
-TEST_MAIN()
-{
-    vespalib::rmdir(test_dir, true);
```

Comment:
Then I guess a more general solution will be to let DirectoryHandler do cleanup of the directory in the constructor. Feel free to fix that. It will help in other tests as well.